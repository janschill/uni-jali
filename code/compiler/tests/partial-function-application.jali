func f x y z =
  k = x + y * z;
  k
end

f 1 2