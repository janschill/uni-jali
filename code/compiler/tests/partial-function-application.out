Closure("part_f",["z"],
  Let("k",Prim("+",Variable"x",Prim("*",Variable"y",Variable"z")),
  Variable"k"),
    [("x",IntegerValue1);
    ("y",IntegerValue2);
    ("f",Closure("f",["x";"y";"z"],
      Let("k",Prim("+",Variable"x",Prim("*",Variable"y",Variable"z")),
      Variable"k"),[]))
    ])